From 864db95568d874164bbda79341f39ffd3bf3428b Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 01 Dec 2021 11:49:16 +0200
Subject: [PATCH] [SW-64797] gaudi2: security: update pcie dev id

Pcie device id is updated to 1030 for secured devices. This
data is used by LKD in order to determine that security is
enabled.

pcie dev id:
Security disabled : 1020
Security enabled : 1030

Change-Id: I6956c2a99933d81b070189afe1e01329ab50f90c
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 3b5afcb..664a02a 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -55,6 +55,7 @@
 #include <pll.h>
 #include <pci.h>
 #include <boot_if.h>
+#include <security.h>
 #include <spi_flash.h>
 
 #define PCIE_LTR_DISABLE		0
@@ -290,10 +291,30 @@
 	return mmPSOC_EFUSE_READ_CTRL;
 }
 
+static u32 pci_dev_id;
+
+void preboot_device_id_init(void)
+{
+	pci_dev_id = PCI_IDS_GAUDI2;
+
+	/*
+	 * Secure pci ids:
+	 * PCI_IDS_GAUDI2_SEC : 0x1030
+	 */
+	/* If security is enabled set the PCI ID accordingly*/
+	if (preboot_security_is_enabled())
+		pci_dev_id = PCI_IDS_GAUDI2_SEC;
+
+	/* using vendor id as a magic number */
+	pci_dev_id |= (PCI_VENDOR_ID_HABANALABS << 16);
+}
+
 u32 preboot_device_id_get(void)
 {
-	/* TODO: remove hard coding */
-	return PCI_IDS_GAUDI2;
+	if ((pci_dev_id >> 16) != PCI_VENDOR_ID_HABANALABS)
+		return 0;
+
+	return pci_dev_id & 0xffff;
 }
 
 static void pcie_update_reg_range(u32 addr, u32 addr_end, u32 value)
